WebApp/autoreduce_webapp/reduction_viewer/view_utils.py

```diff
@@ -22,19 +22,17 @@ def deactivate_invalid_instruments(func):
     Deactivate instruments if they are invalid
     """
 
+    @functools.wraps(func)
     def request_processor(request, *args, **kws):
         """
         Function decorator that checks the reduction script for all active instruments
         and deactivates any that cannot be found
         """
-        # pylint:disable=no-member
-        instruments = Instrument.objects.filter(is_active=True)
+        instruments = Instrument.objects.all()
         for instrument in instruments:
-            reduction_path = os.path.join(REDUCTION_DIRECTORY % instrument.name, 'reduce.py')
-            if not os.path.isfile(reduction_path):
-                LOGGER.warn("Could not find reduction file: %s", reduction_path)
-                instrument.is_active = False
-                instrument.save()
+            instrument.is_active = Path(REDUCTION_DIRECTORY % instrument.name, 'reduce.py').exists()
+            instrument.save()
```

Comment:
Do you know if `.save()` will avoid an actual DB write if the object is unchanged?